[iOS] Enable Diagnostics.Tracing tests

steveisok opened the pull request:
Fixes an issue where a test project that contains native libraries will fool the AOT compiler task in thinking there are assemblies in different locations. This causes .aotdata files to not be bundled with the test app and ends up causing a runtime assertion.

Fixes https://github.com/dotnet/runtime/issues/51382

Commit:
[iOS] Enable Diagnostics.Tracing tests

Fixes an issue where a test project that contains native libraries will fool the AOT compiler task in thinking there are assemblies in different locations. This causes .aotdata files to not be bundled with the test app and ends up causing a runtime assertion.

steveisok commented:
@radical I'm not sure if switching the Filter / Ensure methods are going to impact you.

steveisok commented:
/azp run runtime-extra-platforms

radical commented:
> @radical I'm not sure if switching the Filter / Ensure methods are going to impact you.

I think it should be fine. `FilterAssemblies` does not try to load any dependencies, so location shouldn't matter. But let's see what the tests say :)

radical commented:
/azp run runtime-wasm

mdh1418 commented on src/libraries/tests.proj:
```diff
@@ -311,7 +311,6 @@
 
     <!-- App Crash https://github.com/dotnet/runtime/issues/53624 -->
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj" />
-    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Diagnostics.Tracing/tests/System.Diagnostics.Tracing.Tests.csproj" />
```
We also need to remove the ActiveIssues like https://github.com/dotnet/runtime/blob/bdc0d1a409a24d4092e7b108c13fc4ff1c643d1f/src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestNegative.cs#L62

mdh1418 commented:
We can reenable these as well https://github.com/dotnet/runtime/issues/56073
We just need to add ```RuntimeComponents="$(RuntimeComponents)"``` to `AppleAppBuilderTask` in `AppleApp.targets` and setting this property in `System.Diagnostics.Tracing.Tests.csproj` ```<RuntimeComponents>diagnostics_tracing</RuntimeComponents>```

radical commented:
```
Precompiling failed for /datadisks/disk1/work/A48F08FA/w/A9BB08F7/e/blazorwasm_Release_aot/obj/Release/net7.0/wasm/for-publish/aot-in/Microsoft.AspNetCore.Components.WebAssembly.dll with exit code 1.
Failed to load method 0x6000065 from '/datadisks/disk1/work/A48F08FA/w/A9BB08F7/e/blazorwasm_Release_aot/obj/Release/net7.0/wasm/for-publish/aot-in/Microsoft.AspNetCore.Components.WebAssembly.dll' due to Could not load file or assembly 'Microsoft.JSInterop.WebAssembly, Version=7.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies..
Run with MONO_LOG_LEVEL=debug for more information.
AOT of image Microsoft.AspNetCore.Components.WebAssembly.dll failed.
Mono Ahead of Time compiler - compiling assembly /datadisks/disk1/work/A48F08FA/w/A9BB08F7/e/blazorwasm_Release_aot/obj/Release/net7.0/wasm/for-publish/aot-in/Microsoft.AspNetCore.Components.WebAssembly.dll
```

Hm `Failed to load method 0x6000065 from '/datadisks/disk1/work/A48F08FA/w/A9BB08F7/e/blazorwasm_Release_aot/obj/Release/net7.0/wasm/for-publish/aot-in/Microsoft.AspNetCore.Components.WebAssembly.dll' due to Could not load file or assembly 'Microsoft.JSInterop.WebAssembly, ..`

`Microsoft.JSInterop.WebAssembly` is skipped for AOT, but looks like it is still needed by `mono-aot-cross`. Hm I'll push a fix for this.

radical commented:
> Fixes an issue where a test project that contains native libraries will fool the AOT compiler task in thinking there are assemblies in different locations. This causes .aotdata files to not be bundled with the test app and ends up causing a runtime assertion.

Could you explain what exactly is happening here?
- test project contains native dlls, in location different from the managed assemblies
- which makes the aot compiler task to move all of them (including the native ones) to a single directory
- but how does that affect the `aotdata` being bundled?

radical committed:
Merge remote-tracking branch 'origin/main' into enable-diag-tracing-ios

radical committed:
MonoAOTCompiler: Ensure `.aotdata` files end up next to the original assemblies

`iOS` targets depend on these files being next to the original assemblies. But
that assumption breaks when `MonoAOTCompiler` copies the assemblies to `aot-in`
directory, and emits the `.aotdata` files there.

radical committed:
MonoAOTCompiler: Correctly handle assemblies to skip for aot, and

.. unmanaged assemblies.

1. Completely ignore unmanaged assemblies
2. Assemblies to be skipped for AOT, should still get copied to `aot-in`
dir. If this is not done, then blazor apps would break because they
skip` Microsoft.JSInterop.WebAssembly` but other assemblies depend on
it, thus causing `mono-aot-cross` to fail because it can't find the
JSInterop assembly.

radical commented:
I have pushed some changes that fixes the issue in two ways:

1. The `.aotdata` files will be emitted next to the original assemblies
2. Unmanaged assemblies won't be used to decide whether we need to move all the assemblies to a `aot-in` directory

radical commented:
/azp run runtime-extra-platforms,runtime-wasm

Commit:
Re-enable more iOS tests and set diagnostics_tracing runtime component

steveisok commented:
/azp run runtime-extra-platforms

radical committed:
MonoAOTCompiler: revert hack for .aotdata files on ios

radical committed:
Merge remote-tracking branch 'origin/main' into enable-diag-tracing-ios

radical committed:
MonoAOTCompiler: emit a message instead of a warning when skipping unmanaged assemblies

radical commented:
/azp run runtime-extra-platforms,runtime-wasm

Commit:
Add AppleAppBuilder change that reads path in from AotDataFile item

steveisok commented:
/azp run runtime-extra-platforms

Commit:
Properly enable tracing tests

steveisok commented:
/azp run runtime-extra-platforms

steveisok commented:
System.Diagnostics.Tracing tests pass on tvOS.  The other failures are unrelated.

mdh1418 commented on src/libraries/tests.proj:
```diff
@@ -278,6 +278,19 @@
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Reflection/tests/System.Reflection.Tests.csproj" />
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Threading.Channels/tests/System.Threading.Channels.Tests.csproj" />
 
+    <!-- 
+      System.Formats.Cbor crashes on tvOS 
+    -->
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Formats.Cbor\tests\System.Formats.Cbor.Tests.csproj" />
+
+    <!--
+      Test apps that are too large and take too long to build 
+      Keep here until aggressive trimming targets can work on helix.
+    -->
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.RegularExpressions\tests\UnitTests\System.Text.RegularExpressions.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn4.0.Tests.csproj" />
+
+
```
Are these new failures? Are there issues associated that we can put in the comments?

radical replied:
If it's just links that we need to add, then we can do that in a follow up PR.

steveisok replied:
There will be.  I would rather address it in https://github.com/dotnet/runtime/pull/67861

mdh1418 reviewed (approved):
Looks good to me! If we can add new issues/reference existing ones for any new project/test skips that'd be good.
